src/main/java/com/shipmentEvents/demo/TestBean.java

```diff
@@ -0,0 +1,38 @@
+package com.shipmentEvents.handlers;
+
+import java.sql.Connection;
+import java.sql.Statement;
+import java.sql.ResultSet;
+import javax.naming.Context;
+import javax.naming.InitialContext;
+import javax.sql.DataSource;
+
+public class TestBean {
+
+	public void doBusiness() {
+		Connection con = null;
+		try {
+			Context initContext = new InitialContext();
+			Context envContext  = (Context)initContext.lookup("java:/comp/env");
+			DataSource ds = (DataSource)envContext.lookup("jdbc/oracle");
+			con = ds.getConnection();
+			Statement st = con.createStatement();
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the following resource: *st*. Currently, there are execution paths that do not contain closure statements. Either a) close *st* in a try-finally block or b) close the resource by declaring *st* in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).